docs/content/TextArea.mdx

```diff
@@ -0,0 +1,187 @@
+
+```js
+import {TextArea} from '@primer/react'
+```
+
+## Examples
+
+<Note variant="warning">
+
+TextArea components **must always** be accompanied by a corresponding label to improve support for assistive
+technologies. Examples below are provided for conciseness and may not reflect accessibility best practices.
+
+</Note>
+
+### Basic
+
+```javascript live noinline
+const TextAreaExample = () => {
+  // Running in controlled mode (recommended)
+  const [value, setValue] = React.useState('')
+
+  const handleChange = event => {
+    setValue(event.target.value)
+  }
+
+  return <TextArea onChange={handleChange}>{value}</TextArea>
+}
+
+render(<TextAreaExample />)
+```
+
+### Uncontrolled mode
+
+```javascript live noinline
+const TextAreaExample = () => {
+  const ref = React.useRef()
+
+  const handleSubmit = event => {
+    event.preventDefault()
+    if (!ref.current.value) {
+      alert(`Enter a value into the TextArea and press submit`)
+      return
+    }
+
+    alert(`Current TextArea value: ${ref.current.value}`)
+  }
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <TextArea ref={ref} defaultValue="Set the initial state in uncontrolled mode using the defaultValue prop" />
+      <br />
+      <Button type="submit">Submit</Button>
+    </form>
+  )
+}
+
+render(<TextAreaExample />)
+```
+
+### Displaying form validation state
+
+```jsx live
+<>
+  <Box as="label" for="success-state" sx={{display: 'block'}}>
+    Success state:
+  </Box>
+  <TextArea id="success-state" validationStatus="success" />
+  <Box as="label" for="error-state" sx={{display: 'block', mt: 5}}>
+    Error state:
+  </Box>
+  <TextArea id="error-state" validationStatus="error" />
+</>
+```
+
+### Inactive
+
+```jsx live
+<>
+  <TextArea disabled>This text is inactive</TextArea>
+</>
+```
+
+### Resizing and constraining dimensions
+
+By default, `TextArea` can be resized by the user vertically and horizontally.
+
+Resizing can be prevented by setting `resize` to `none`, with `height` and `width` props available for granular control of its appearance.
+Setting an explicit `height` or `width` can prevent its respective `resize` options from being applied.
+
+```jsx live
+<TextArea resize="none" height={400}></TextArea>
+```
+
+### Custom styling
+
+```jsx live
+<>
+  <TextArea sx={{marginBottom: 2}}></TextArea>
+  <p>Custom styles like `margin` and `padding` can be applied using the `sx` prop</p>
+</>
+```
+
+## Props
+
+### TextArea
+
+<PropsTable>
+  <PropsTableRow
+    defaultValue="undefined"
+    name="required"
+    type="boolean"
+    description="Indicates to the user and assistive technologies that the field value is required"
+  />
+  <PropsTableRow
+    name="block"
+    type="boolean"
+    defaultValue="false"
+    description="Expands with width of the component to fill the parent elements"
+  />
+  <PropsTableRow
+    name="block"
+    type="boolean"
+    defaultValue="false"
+    description={<>Creates a full width input element</>}
```

Comment:
1. Why does this need to be in a fragment?
2. Let's use this description in the JSDoc comment in the TextArea's `Props` type

Reply:
👍 it doesn't, will remove it. I was copy/pasting too fast 😓